review-notifications/src/Github.js

```
@@ -1,38 +1,222 @@
       });
+    });
+    this.setState({
+      createdPR: createdPR,
+      assignedPR: assignedPR,
+      mentionedPR: mentionedPR,
+      reviewedPR: reviewedPR,
+    });
   }
-  listOfTitles() {
-    const list = this.state.pulls.map(pr => <li key={pr.id}>{pr.title}</li>);
-    return <ul>{list}</ul>;
+
+  async getPullRequests() {
+    let query = `https://api.github.com/users/${this.state.user}/repos`;
+    if (this.state.token !== '' && this.state.auth) {
+      query = `https://api.github.com/user/repos?access_token=${
+        this.state.token
+      }`;
+    }
+    let prLinksList = [];
+    try {
+      const response = await axios.get(query);
+      prLinksList = response.data.map(repo =>
+        repo.pulls_url.replace('{/number}', '')
+      );
+    } catch (error) {
+      console.log(error);
+    }
+    try {
+      const pullRequests = await axios.all(
+        prLinksList.map(prLink =>
+          axios.get(`${prLink}?access_token=${this.state.token}`)
+        )
+      );
+      const prData = await Promise.all(
+        pullRequests.map(
+          async prList =>
+            await Promise.all(
+              prList.data.map(
+                async pullRequest => await this.extractDataFromPR(pullRequest)
+              )
+            )
+        )
+      );
+      this.filterPullRequests(prData);
+      this.setState({ haveData: true });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  listOfPullRequest() {
+    let list = [];
+    this.state.prOptions.forEach(option => {
+      if (option.isChecked)
+        switch (option.value) {
+          case 'Created':
+            list.push(...this.state.createdPR);
+            break;
+          case 'Assigned':
+            list.push(...this.state.assignedPR);
+            break;
+          case 'Mentioned':
+            list.push(...this.state.mentionedPR);
+            break;
+          case 'Review request':
+            list.push(...this.state.reviewedPR);
+            break;
```

You should probably add a `default` case just for safety. Maybe throw an error in there or simply do nothing